Assets/MRTK/Services/InputSystem/MixedRealityInputModule.cs

```diff
@@ -305,8 +305,15 @@ protected PointerEventData.FramePressState StateForPointer(PointerData pointerDa
         void IMixedRealityPointerHandler.OnPointerUp(MixedRealityPointerEventData eventData)
         {
             int pointerId = (int)eventData.Pointer.PointerId;
-            Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId));
-            pointerDataToUpdate[pointerId].nextPressState = PointerEventData.FramePressState.Released;
+
+            // OnPointerUp can be raised during an OnSourceLost. If the pointer has already been removed
+            // from the pointerDataToUpdate Dictionary and added to the pointerDataToRemove list, then
+            // that pointer's state update can be ignored.
+            Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId) || IsPointerIdInRemovedList(pointerId));
```

Comment:
We could probably get rid of this - I'm guessing that @matthejo did this because removing an assert in code that you don't directly own is a little scarier than augmenting it to allow for a specific case that you are aware that can happen